enhancements/deprecated-bundles.md

```diff
@@ -0,0 +1,131 @@
+
+- editing the `Subscription`'s `channel` field to start receiving updates from a different channel
+- creating a custom `CatalogSource` pointing to the previous OCP index, editing the `Subscription`'s `source` field to point to the new `CatalogSource` in order to receive updates.
+
+With the new work on the skip range being steppable, the above scenario may be less likely to happen.
+
+A new field will be added to the operator-registry database table `operatorbundle` called `deprecated` and the bundle images specified in the command will be marked as such.
+
+The OLM resolver will be modified with a new property type called `olm.deprecated` in order to filter out any options that have this property set and disallow installs of deprecated bundles. This also means appropriate status reporting in the subscription and installplan need to be added to OLM, and alerts will be fired if a deprecated version is installed.
+
+### User Stories
+
+#### Story 1
+
+As an index maintainer I want to be able to truncate update graphs in order to remove or prevent installs of incompatible or unsupported versions of certain operators.
+
+##### Implementation Details
+
+A new field is added to the operator-registry database table `operatorbundle` called `deprecated` and the bundle images specified in the command will be marked as such. Versions that come after the deprecated version in the update graph (in all channels) are removed from the database.
+
+For example, suppose following channel:
+
+```
+    1.4.0 -- replaces -> 1.3.0 -- replaces -> 1.2.0 -- replaces -> 1.1.0
+```
+
+we then deprecate version 1.3.0 using the following command
+
+```
+    $ opm index deprecate --bundles "quay.io/my/bundle:1.3.0" --from-index "quay.io/my/index:v4.6" --tag "quay.io/my/index:v4.7"
+```
+
+Now, in `quay.io/my/index:v4.7` the above channel looks like:
+
+```
+    1.4.0 -- replaces -> 1.3.0 [deprecated]
+```
+
+Deprecated versions are upgradable but not installable. To achieve this, we will modify the operator-registry apis used by OLM to determine what can be installed.
+
+#### Story 2
+
+As an index maintainer I want to be able to undo the above operation ideally without removing the entire package and re-adding all the bundles.
+
+##### Implementation Details
+
+The `opm index rm` command is modified to allow removal of the tail of a channel (deprecated versions will always lie at the tail of the graph).
+
+The `opm index add` command is modified to allow adding to the tail of a channel (removed versions from the deprecate command will always have been from the tail)
+
+With these two additions we can now undo the above deprecation by:
```

Comment:
Why do we need to undo at all? If we're really in such a bad state that reverting to an old index isn't feasible and we need to do surgery on the index, don't we already have this by just index rm'ing the entire package and re-adding the set of bundles from scratch?

Reply:
yes that's just a painful UX but this is probably ok in the short term